### PR TITLE
[DPE-2071] Refactoring: converging code with k8s charm

### PR DESCRIPTION
**Pull request by dmitry-ratushnyy**
## Issue
After some refactoring done in MongoDB K8S charm the code base  of charms diverged. 

## Solution
The goal of this pull request is to converge the code with MongoDB K8s charm

### CODE REVIEW

**Review by delgod — APPROVED** [review]

File: src/machine_helpers.py [hunk 1]
```diff
@@ -56,3 +56,14 @@ def push_file_to_unit(parent_dir, file_name, file_contents) -> None:
...
+
+def delete_file_on_unit(parent_dir, file_name) -> None:
+    """Removes a file from unit."""
+    file_path = os.path.join(parent_dir, file_name)
+
+    if os.path.isfile(file_path):
+        os.remove(file_path)
+        logger.debug(f"File {file_name} deleted from {parent_dir}.")
+    else:
+        logger.error(f"Errof delleting not existing file {file_path}")
```
Comment (Mehdi-Bendriss) [inline src/machine_helpers.py:69]: Why is this necessary? If the file doesn't exist the code should just move on

File: src/charm.py [hunk 2]
```diff
@@ -824,6 +827,34 @@ def _push_tls_certificate_to_workload(self) -> None:
...
+            delete_file_on_unit(
+                parent_dir=Config.MONGOD_CONF_DIR,
+                file_name=TLS_INT_CA_FILE,
+            )
+
+        if internal_pem is not None:
+            delete_file_on_unit(
+                parent_dir=Config.MONGOD_CONF_DIR,
+                file_name=TLS_INT_PEM_FILE,
+            )
```
Comment (Mehdi-Bendriss) [inline src/charm.py:856]: Can't this be simplified to this? since you already test on the existence of the file on `delete_file_on_unit`

```suggestion
        delete_file_on_unit(parent_dir=Config.MONGOD_CONF_DIR, file_name=TLS_EXT_CA_FILE)
        delete_file_on_unit(parent_dir=Config.MONGOD_CONF_DIR, file_name=TLS_EXT_PEM_FILE)
        delete_file_on_unit(parent_dir=Config.MONGOD_CONF_DIR, file_name=TLS_INT_CA_FILE)
        delete_file_on_unit(parent_dir=Config.MONGOD_CONF_DIR, file_name=TLS_INT_PEM_FILE)
```

Comment (delgod) [reply src/charm.py]: fixed in https://github.com/canonical/mongodb-operator/pull/234/commits/9457f636d7faa7778bb4d7d9e1290e0bc34670fe

File: src/machine_helpers.py (hunk 1 above)
Comment (delgod) [reply src/machine_helpers.py]: fixed https://github.com/canonical/mongodb-operator/pull/234/commits/e49aac7289eba2a9fbd5eaa0b8e70fe838099660

**Review by delgod — APPROVED** [review]

**Review by Mehdi-Bendriss — DISMISSED** [review]

**Review by delgod — APPROVED** [review]
